### DISCUSSION TITLE
CandleStick Chart, Suppress Missing X Values

### DISCUSSION THREAD

**Discussion by my-waiwai** [patrykandpatrick/vico#956] [Questions] [answered]
### Question

I've been working on an app that works with US stock market data and have noticed that the Candlestick chart just doesn't work very well when the data spans more than the most recent 1 day of data. Displaying single-day data the chart displays all data points smoothly and it looks excellent. Once I ask to display 1 week of data, more than 75% of the chart is just blank. This makes perfect sense when you consider the market is only open 6.5 hours per day and the chart is displaying the full range of dates/times across 7 full days (including nights and weekends when the market is closed).

As it stands right now Friday's 6.5 hours worth of data is followed by 65.5 hours worth of blank space. Not only does this mean the chart scrolls left-right for a _**long**_ way it's more than 75% empty space and means you can't really compare price movements across multiple days very effectively. This gets increasingly noticeable when the period extends to 1 month or more. Below is a screenshot of the 1 week data using a 1 hour interval. If I choose a 15 minute interval, the Nov 21 data gets pushed off the screen to the left. While I can scroll to see it, the gap between the candlesticks groups take up nearly the entire width of the screen (that gap is just the overnight hours).

![image](https://github.com/user-attachments/assets/fdb1adb2-7120-4415-8b33-251488ed7d31)

X values are timestamps (seconds since epoch) and I only provide the `candlestickSeries` with the actual data points, not the gaps.

Is there a suggestion for how to hide/suppress/not show the missing data points on the chart so that Friday's data squishes right up next to Monday's? This would make the chart much more compact and much more usable for multi-day/week/month data sets.

I'm also using the LineChart and I'd love to be able to suppress the gaps on that chart, as well, rather than a straight line covering the gaps. The connecting lines are very jarring.

Thanks in advance for any advice or help!

### Vico version(s)

2.0.0-beta.2

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello, @my-waiwai. Yes, this is possible.

Entries are positioned in a mathematically accurate manner. If you need the last candle for a given day and the next candle for the next day to be as far apart as two neighboring candles for the same day, then the _x_ deltas should be the same. Please update your _x_ values accordingly and modify the formatter as required to preserve the labeling. The simplest approach is to use consecutive integers as the _x_ values.

You can use the same solution for line charts.

**Reply by my-waiwai** [reply]
@Gowsky Astonishingly simple to make this work. I replaced the _x_ values with an integer sequence and updated my `CartesianValueFormatter` to use the _x_ value as an index into the timestamp list to get the date which I was already formatting in a human-readable way. I was over-complicating the issue and completely overlooked this solution.

Excellent library and thanks for the support!
